### PR TITLE
Update Self-Diagnostics troubleshooting readme

### LINKED ISSUE #2548: Self-diagnostics log file has strange content
# Bug Report

I am using OpenTelemetry 1.1.0 (commit `ce46d001d5ed08d91e7750b4859410330d57e7a7`). Running [AspNetCore example] (https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/AspNetCore) with target framework `netcoreapp3.1`, release configurations and `OTEL_DIAGNOSTICS.json` config file:

```
{
    "LogDirectory": ".",
    "FileSize": 1024,
    "LogLevel": "LogAlways"
}
```

After some requests to https://localhost:5001/WeatherForecast/ the log file `Examples.AspNetCore.exe.41136.log` was created.

[Examples.AspNetCore.exe.41136.log](https://github.com/open-telemetry/opentelemetry-dotnet/files/7441237/Examples.AspNetCore.exe.41136.log)

When I open .log file in Notepad++ I see there internal events generated by OpenTelemetry (logs) with some unsupported symbols:

image

But I am expecting to see there **only** logs.

### PR DESCRIPTION
**Pull request by xiang17**
Fixes #2548.

## Changes

Please provide a brief description of the changes here.

* Added description that the file is circularly overwritten
* And thus it might not be a proper text file format
* Arranged a few sentences around for better readability

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry/README.md [hunk 1]
```diff
@@ -329,8 +329,11 @@ this SDK also ships a "self-diagnostics feature", which helps troubleshooting.
 When enabled, internal events generated by OpenTelemetry will be written to a
 log file.
 
-The self-diagnostics feature can be enabled/disabled while the process
-is running.
+The self-diagnostics feature can be enabled/changed/disabled while the process
+is running. The SDK will attempt to read the configuration file every 10 seconds
+in non-exclusive read-only mode. The SDK will create or overwrite a file
+with new logs according to the configuration. This file will not exceed the
+configured max size and will be circularly overwritten.
```
Comment (reyang) [inline src/OpenTelemetry/README.md:336]: Maybe reword "circularly overwritten" to "overwritten in a circular way"?

File: src/OpenTelemetry/README.md [hunk 2]
```diff
@@ -379,19 +385,19 @@ A `FileSize`-KiB log file named as `ExecutableName.ProcessId.log` (e.g.
...
-content will be read.
+log file would be generated.
+
+When the `LogDirectory` or `FileSize` is found to be changed, the SDK will create
+or overwrite a file with new logs according to the new configuration. The
+configuration file has to be no more than 4 KiB. In case the file is larger than
+4 KiB, only the first 4 KiB of content will be read.
+
+The log file might not be a proper text file format to achieve the goal of having
+minimal overhead and bounded resource usage: it may have trailing `NULL`s if log
```
Comment (reyang) [inline src/OpenTelemetry/README.md:398]: Probably `NUL` instead of `NULL`? https://www.rapidtables.com/code/text/ascii-table.html

File: src/OpenTelemetry/README.md [hunk 3]
```diff
@@ -329,8 +329,11 @@ this SDK also ships a "self-diagnostics feature", which helps troubleshooting.
 When enabled, internal events generated by OpenTelemetry will be written to a
 log file.
 
-The self-diagnostics feature can be enabled/disabled while the process
-is running.
+The self-diagnostics feature can be enabled/changed/disabled while the process
+is running. The SDK will attempt to read the configuration file every 10 seconds
```
Comment (reyang) [inline src/OpenTelemetry/README.md:333]:
```suggestion
is running. The SDK will attempt to read the configuration file every `10` seconds
```

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/README.md [hunk 4]
```diff
@@ -362,7 +365,10 @@ You can also find the exact directory by calling these methods from your code.
    can be an absolute path or a relative path to the current directory.
 
 2. `FileSize` is a positive integer, which specifies the log file size in
-   [KiB](https://en.wikipedia.org/wiki/Kibibyte).
+   [KiB](https://en.wikipedia.org/wiki/Kibibyte). This value must be between 1 MiB
+   and 128 MiB (inclusive), or it will be rounded to the closest upper or lower
+   limit. The log file will never exceed this configured size, and will be
+   rewriten in a circular way.
```
Comment (cijothomas) [inline src/OpenTelemetry/README.md:371]:
```suggestion
   rewritten in a circular way.
```

Comment (reyang) [reply src/OpenTelemetry/README.md:371]: I guess overwritten is more accurate than rewritten?

Comment (reyang) [reply src/OpenTelemetry/README.md:371]: Or just "will be written in a circular way."

Comment (cijothomas) [reply src/OpenTelemetry/README.md:371]: overwritten sounds good.

Comment (xiang17) [reply src/OpenTelemetry/README.md]: Updated to overwritten.
